src/main/java/org/powerflows/dmn/io/yaml/YamlDecisionConverter.java

```diff
@@ -38,29 +42,56 @@
 import java.util.stream.Collectors;
 
 public class YamlDecisionConverter implements DecisionToExternalModelConverter<YamlDecision> {
+
     @Override
     public YamlDecision to(final Decision decision) {
+        final ExpressionType decisionExpressionType = findDecisionExpressionType(decision);
+        final EvaluationMode decisionEvaluationMode = Decision.DEFAULT_EVALUATION_MODE.equals(decision.getEvaluationMode()) ? null : decision.getEvaluationMode();
+        final HitPolicy decisionHitPolicy = Decision.DEFAULT_HIT_POLICY.equals(decision.getHitPolicy()) ? null : decision.getHitPolicy();
+
         final YamlDecision yamlDecision = new YamlDecision();
         yamlDecision.setId(decision.getId());
         yamlDecision.setName(decision.getName());
-        yamlDecision.setExpressionType(decision.getExpressionType());
-        yamlDecision.setEvaluationMode(decision.getEvaluationMode());
-        yamlDecision.setHitPolicy(decision.getHitPolicy());
-        yamlDecision.setFields(createFields(decision.getInputs(), decision.getOutputs()));
+        yamlDecision.setExpressionType(Decision.DEFAULT_EXPRESSION_TYPE.equals(decisionExpressionType) ? null : decisionExpressionType);
+        yamlDecision.setEvaluationMode(decisionEvaluationMode);
+        yamlDecision.setHitPolicy(decisionHitPolicy);
+        yamlDecision.setFields(createFields(decision.getInputs(), decision.getOutputs(), decisionEvaluationMode, decisionExpressionType));
 
         final Map<String, Input> inputsMap = decision.getInputs()
                 .stream()
                 .collect(Collectors.toMap(Input::getName, Function.identity()));
 
         yamlDecision.setRules(decision.getRules()
                 .stream()
-                .map(rule -> ruleToYamlRule(rule, inputsMap))
+                .map(rule -> ruleToYamlRule(rule, inputsMap, decisionExpressionType))
                 .collect(Collectors.toList()));
 
         return yamlDecision;
     }
 
-    private YamlFields createFields(final List<Input> inputs, final List<Output> outputs) {
+    private ExpressionType findDecisionExpressionType(final Decision decision) {
+        final ExpressionType expressionType = decision.getRules().get(0).getInputEntries().get(0).getExpression().getType();
```

Comment:
Because in this code I want to check if all entries have the same expression type. And I can safely  use index 0 because of decision validator in its builder - doesn't allow to create the object if no entries.